### PR TITLE
[Imaging Uploader] Prevent upload duplication and handle other mriFile statuses

### PR DESCRIPTION
**Pull request by davidblader**
https://redmine.cbrain.mcgill.ca/issues/13190
https://redmine.cbrain.mcgill.ca/issues/13194
When retrieving the list of MRI's, _getFileList will only append files which already exist in the incoming data directory. However these files get deleted when the MRI pipeline runs successfully, allowing for duplicate uploads for previously successful files. It can be assumed that the insertion was successful based on the number of MINCs inserted being > 0. Use this in addition to file existence to generate the mriList which gets checked when an upload form is submitted

### CODE REVIEW

File: modules/imaging_uploader/jsx/UploadForm.js [hunk 1]
```diff
@@ -101,6 +101,17 @@ class UploadForm extends React.Component {
       return;
     }
 
+    // File in the middle of insertion pipeline
+    if (mriFile.status === "In progress") {
```
Comment (MounaSafiHarab) [inline modules/imaging_uploader/jsx/UploadForm.js]: `In Progress`; it is case sensitive I think

**Comment by MounaSafiHarab** [comment]
@davidblader 

works for me now;
for the In Progress... case, I get the correct error message after I refresh Imaging Uploader page only 

imaguploader

**Comment by MounaSafiHarab** [comment]
@driusan 

I am assuming this fix can not make it into 17.1; it has to be in 17.2?

**Comment by driusan** [comment]
Why can't it go into 18.0.1? (Or whatever the .z is by the time the backlog of open PRs is by the time it catches up to this..)

**Comment by MounaSafiHarab** [comment]
Okay @driusan 
it is a bug fix, so I hope it can make it into the next bug fix release.

File: modules/imaging_uploader/php/NDB_Menu_Filter_imaging_uploader.class.inc [hunk 2]
```diff
@@ -596,14 +596,12 @@ class NDB_Menu_Filter_Imaging_Uploader extends NDB_Menu_Filter_Form
      */
     function _getFileList($data)
     {
-
         $files = array();
-
         foreach ($data as $row) {
+            $uploadStatus = $row[1];
             $filePath     = $row[5];
+            $mincInserted = $row[10];
```
Comment (ridz1208) [inline modules/imaging_uploader/php/NDB_Menu_Filter_imaging_uploader.class.inc]: @davidblader is there a risk of these hard coded values being different from a project to another ? if so, a redmine ticket should be issued in parallell to flag this for change in a minor version

**Comment by ridz1208** [comment]
@driusan to the limited extent of my knowledge, this looks good.but make sure to review it more thoroughly since its on the MRI side of things

**Comment by driusan** [comment]
@davidblader This has conflicts that need to be resolved

**Pull request merged**